src/test/java/org/example/jvspringbootfirstbook/service/cart/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,312 @@
+
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
+    @Mock
+    private CartItemRepository cartItemRepository;
+    @Mock
+    private CartItemMapper cartItemMapper;
+    @Mock
+    private BookRepository bookRepository;
+
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @Test
+    @DisplayName("""
+            Find user cart with correct user returns ShoppingCartDto
+            """)
+    void findUserCart_withCorrectUser_returnsShoppingCartDto() {
+        User user = getUser();
+        ShoppingCart shoppingCart = getShoppingCart(user);
+        ShoppingCartDto expected = getShoppingCartDto(user, Set.of());
+
+        when(shoppingCartRepository.findShoppingCartByUser(user.getId()))
+                .thenReturn(shoppingCart);
+        when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(expected);
+
+        ShoppingCartDto actual = shoppingCartService.findUserCart(user);
+
+        Assertions.assertAll(
+                () -> Assertions.assertEquals(expected, actual),
+                () -> Assertions.assertEquals(user.getId(), actual.userId())
+        );
+
+        verify(shoppingCartRepository, times(1)).findShoppingCartByUser(user.getId());
+        verify(shoppingCartMapper, times(1)).toDto(shoppingCart);
+        verifyNoMoreInteractions(shoppingCartRepository, shoppingCartMapper);
+    }
+
+    @Test
+    @DisplayName("""
+            Add item with valid input returns ShoppingCartDto
+            """)
+    void addItem_ValidInput_returnsShoppingCartDto() {
+        User user = getUser();
+        ShoppingCart shoppingCart = getShoppingCart(user);
+
+        Book book = getBook();
+        int quantity = 1;
+
+        CartItemRequestDto cartItemRequestDto = getCartItemRequestDto(quantity, book);
+        CartItem cartItem = getCartItem(shoppingCart, book, cartItemRequestDto);
+        CartItemDto cartItemDto = getCartItemDto(book, (long) quantity);
+        Set<CartItemDto> itemDtoSet = Set.of(cartItemDto);
+        ShoppingCartDto expected = getShoppingCartDto(
+                user, itemDtoSet);
+
+        when(cartItemMapper.toModel(cartItemRequestDto)).thenReturn(cartItem);
+        when(bookRepository.findById(book.getId())).thenReturn(Optional.of(book));
+        when(shoppingCartRepository.findShoppingCartByUser(user.getId())).thenReturn(shoppingCart);
+        when(shoppingCartRepository.save(shoppingCart)).thenReturn(shoppingCart);
+        when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(expected);
+
+        ShoppingCartDto actual = shoppingCartService.addItem(user, cartItemRequestDto);
+
+        Assertions.assertAll(
+                () -> Assertions.assertNotNull(actual),
+                () -> Assertions.assertEquals(expected, actual),
+                () -> Assertions.assertEquals(user.getId(), actual.userId())
+        );
```

Comment:
Why are you wraping it in `Assertions.assertAll(...)`? This is redundant, you can simply call them one after another.

Reply:
Same applies to all places where you used it.